extratrees/src/main/java/binnie/extratrees/machines/fruitpress/FruitPressLogic.java

```diff
@@ -44,6 +44,9 @@ public ErrorState canProgress() {
 
 	@Override
 	protected void onFinishTask() {
+		final FluidStack tankFinish = FruitPressRecipeManager.getOutput(this.getUtil().getStack(FruitPressMachine.SLOT_CURRENT));
+		tankFinish.amount /= 25;
+		this.getUtil().fillTank((FruitPressMachine.TANK_OUTPUT), tankFinish);
```

Comment:
After looking at the code a bit more I don't think this is the correct fix. The method `onUpdate` should output the correct amount over all the cycles but it looks like there are rounding/arithmetic errors. So I think the fix is in that method not this one.

Or maybe I'm missing something about how the fruit press works

Reply:
You know, I was unsure of that fix. It was a bit hacky to be sure. I'll take another look at that and make it better.

Reply:
That should be much better.